portfolio/src/main/webapp/script.js

```diff
@@ -119,18 +119,62 @@ document.addEventListener('keydown', function(event) {
 
 /** Fetches Comments with specified Max Number of Comments */
 function getComments(maxComments) {
+
   const url = "/comments?maxcomments=" + maxComments;
+
   fetch(url).then(response => response.json()).then((comments) => {
+
     const commentsHistory = document.getElementById('comments-history');
     commentsHistory.innerHTML = '';
-    comments.forEach(message => {
-        var listNode = document.createElement('li');
-        listNode.innerText = message;
-        commentsHistory.appendChild(listNode);
+
+    comments.forEach(comment => {
+        var commentNode = fetchCommentsHelper(comment);
+        commentsHistory.appendChild(commentNode);
```

Comment:
This pattern here of applying a function to each element of an array and putting the result into a new array is called mapping. If you want an extra challenge, check out `Array.map()`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map. Because it returns an array, you can call `.map()` on the result, as well, and chain them.

Reply:
I'll definitely look into this tomorrow! :)

Reply:
Done! I ended up chaining `.map()` with `forEach()` since I didn't need an array returned on the second portion.